// tg-modal/dist/remote/TimedCSSTransitionGroup.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _validateTransitionProp = require('./validateTransitionProp');

var _validateTransitionProp2 = _interopRequireDefault(_validateTransitionProp);

var _TimedCSSTransitionGroupChild = require('./TimedCSSTransitionGroupChild');

var _TimedCSSTransitionGroupChild2 = _interopRequireDefault(_TimedCSSTransitionGroupChild);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TimedCSSTransitionGroup = function (_Component) {
    (0, _inherits3.default)(TimedCSSTransitionGroup, _Component);

    function TimedCSSTransitionGroup() {
        var _Object$getPrototypeO;

        var _temp, _this, _ret;

        (0, _classCallCheck3.default)(this, TimedCSSTransitionGroup);

        for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
            args[_key] = arguments[_key];
        }

        return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_Object$getPrototypeO = (0, _getPrototypeOf2.default)(TimedCSSTransitionGroup)).call.apply(_Object$getPrototypeO, [this].concat(args))), _this), _this._wrapChild = function (child) {
            var _this2 = _this;
            var props = _this2.props;


            return _react2.default.createElement(
                _TimedCSSTransitionGroupChild2.default,
                {
                    name: props.transitionName,
                    appear: props.transitionAppear !== false ? props.transitionAppear : false,
                    enter: props.transitionEnter !== false ? props.transitionEnter : false,
                    leave: props.transitionLeave !== false ? props.transitionLeave : false,

                    afterAppear: props.afterAppear,
                    afterEnter: props.afterEnter,
                    afterLeave: props.afterLeave
                },
                child
            );
        }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
    }

    (0, _createClass3.default)(TimedCSSTransitionGroup, [{
        key: 'render',
        value: function render() {
            // remove warning for unused vars / missing proptype definition
            /* eslint no-unused-vars: 0, react/prop-types: 0 */
            var _props = this.props;
            var afterEnter = _props.afterEnter;
            var afterLeave = _props.afterLeave;
            var animating = _props.animating;
            var rest = (0, _objectWithoutProperties3.default)(_props, ['afterEnter', 'afterLeave', 'animating']);

            return _react2.default.createElement(require('react-addons-transition-group'), (0, _extends3.default)({}, rest, { childFactory: this._wrapChild }));
        }
    }]);
    return TimedCSSTransitionGroup;
}(_react.Component); /**
                      * This is adapted from Facebook's ReactCSSTransitionGroup which is in the React
                      * addons and under the BSD License.
                      */

TimedCSSTransitionGroup.displayName = 'TimedCSSTransitionGroup';
TimedCSSTransitionGroup.propTypes = {
    transitionName: _TimedCSSTransitionGroupChild2.default.propTypes.name,

    transitionAppear: _validateTransitionProp2.default,
    transitionEnter: _validateTransitionProp2.default,
    transitionLeave: _validateTransitionProp2.default,

    afterAppear: _react.PropTypes.func,
    afterEnter: _react.PropTypes.func,
    afterLeave: _react.PropTypes.func
};
TimedCSSTransitionGroup.defaultProps = {
    transitionAppear: false,
    transitionEnter: 1000,
    transitionLeave: 1000
};
exports.default = TimedCSSTransitionGroup;
