// tg-modal/dist-013/components/dom/ModalDialog.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ModalDialog = function (_Component) {
    (0, _inherits3.default)(ModalDialog, _Component);

    function ModalDialog() {
        var _Object$getPrototypeO;

        var _temp, _this, _ret;

        (0, _classCallCheck3.default)(this, ModalDialog);

        for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
            args[_key] = arguments[_key];
        }

        return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_Object$getPrototypeO = (0, _getPrototypeOf2.default)(ModalDialog)).call.apply(_Object$getPrototypeO, [this].concat(args))), _this), _this.onCancel = function (e) {
            if (e && e.preventDefault) {
                e.preventDefault();
            }

            _this.props.onCancel(e, null);
        }, _this.stopPropagate = function (e) {
            e.stopPropagation();
        }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
    }

    (0, _createClass3.default)(ModalDialog, [{
        key: 'render',
        value: function render() {
            var _props = this.props;
            var children = _props.children;
            var isBasic = _props.isBasic;

            var className = 'modal' + (isBasic ? ' modal-basic' : '');

            return _react2.default.createElement(
                'div',
                { className: className, onClick: this.onCancel },
                _react2.default.createElement(
                    'div',
                    { className: 'modal-dialog' },
                    _react2.default.createElement(
                        'div',
                        { className: 'modal-content', onClick: this.stopPropagate },
                        children
                    )
                )
            );
        }
    }]);
    return ModalDialog;
}(_react.Component);

ModalDialog.propTypes = {
    children: _react.PropTypes.node,

    onCancel: _react.PropTypes.func.isRequired,
    isBasic: _react.PropTypes.bool,
    animating: _react.PropTypes.bool
};
exports.default = ModalDialog;
