// tg-modal/dist/components/dom/ModalHeader.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ModalHeader = function (_Component) {
    (0, _inherits3.default)(ModalHeader, _Component);

    function ModalHeader() {
        var _Object$getPrototypeO;

        var _temp, _this, _ret;

        (0, _classCallCheck3.default)(this, ModalHeader);

        for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
            args[_key] = arguments[_key];
        }

        return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_Object$getPrototypeO = (0, _getPrototypeOf2.default)(ModalHeader)).call.apply(_Object$getPrototypeO, [this].concat(args))), _this), _this.onCancel = function (e) {
            if (e && e.preventDefault) {
                e.preventDefault();
            }

            if (!_this.props.isStatic) {
                if (_this.props.onCancel) {
                    _this.props.onCancel(e, null);
                }
            }
        }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
    }

    (0, _createClass3.default)(ModalHeader, [{
        key: 'render',
        value: function render() {
            var _props = this.props;
            var children = _props.children;
            var addClose = _props.addClose;


            var closeBtn = addClose ? _react2.default.createElement(
                'button',
                { className: 'close', 'aria-label': 'Close', onClick: this.onCancel },
                _react2.default.createElement(
                    'span',
                    { 'aria-hidden': 'true' },
                    '×'
                )
            ) : null;

            if (typeof children !== 'string') {
                return children;
            }

            return _react2.default.createElement(
                'div',
                { className: 'modal-header' },
                _react2.default.createElement(
                    'h1',
                    { className: 'modal-title' },
                    children
                ),
                closeBtn
            );
        }
    }]);
    return ModalHeader;
}(_react.Component);

ModalHeader.displayName = 'Modal.Header';
ModalHeader.propTypes = {
    children: _react.PropTypes.node,
    isStatic: _react.PropTypes.bool,
    addClose: _react.PropTypes.bool,
    onCancel: _react.PropTypes.func
};
ModalHeader.defaultProps = {
    addClose: true
};
exports.default = ModalHeader;
